libmamba/src/core/history.cpp

```diff
@@ -325,4 +326,186 @@ namespace mamba
             out << specs_output("neutered", entry.neutered);
         }
     }
+
+    namespace detail
+    {
+        // The following function parses the different formats that can be found in the history
+        // file. conda/mamba1 format: installed: +conda-forge/linux-64::xtl-0.8.0-h84d6215_0
+        // removed: -conda-forge/linux-64::xtl-0.8.0-h84d6215_0 mamba2 broken format: installed:
+        // +conda-forge::xtl-0.8.0-h84d6215_0 removed:
+        // -https://conda.anaconda.org/conda-forge/linux-64::xtl-0.8.0-h84d6215_0 mamba2 new format:
+        // installed: +https://conda.anaconda.org/conda-forge/linux-64::xtl-0.8.0-h84d6215_0
+        // removed: -https://conda.anaconda.org/conda-forge/linux-64::xtl-0.8.0-h84d6215_0
+        specs::PackageInfo pkg_info_builder(std::string s)
+        {
+            std::string s_pkg;
+            std::string channel;
+            std::size_t pos_0 = s.rfind("/");
+            if (pos_0 != std::string::npos)
+            {
+                std::string s_begin = s.substr(0, pos_0);
+                std::string s_end = s.substr(pos_0 + 1, s.size());
+                std::size_t pos = s_begin.rfind("/");
+                if (pos != std::string::npos)
+                {
+                    channel = s_begin.substr(pos + 1, s_begin.size());
+                }
+                else
+                {
+                    channel = s_begin;
+                }
+                s_pkg = std::get<1>(util::rsplit_once(s_end, "::"));
+            }
+            else
+            {
+                channel = std::get<0>(util::split_once(s, "::"));
+                s_pkg = std::get<1>(util::rsplit_once(s, "::"));
```

Comment:
Is it only possible to split once?

```suggestion
                auto double_colon_split = util::split_once(s, "::");

                channel = std::get<0>(double_colon_split);
                s_pkg = std::get<1>(double_colon_split);
```

Note that [a `namespace` field between the two `:` might be included in the future](https://docs.conda.io/projects/conda/en/stable/dev-guide/api/conda/models/match_spec/index.html).

Reply:
Yes, I don't think there can be two `::`.